reposerver/src/main/scala/com/advancedtelematic/tuf/reposerver/http/RepoResource.scala

```diff
@@ -247,11 +262,20 @@ class RepoResource(keyserverClient: KeyserverClient, namespaceValidation: Namesp
         } ~
         (pathEnd & put & entity(as[SignedPayload[TargetsRole]])) { signedPayload =>
           extractRoleChecksumHeader { checksum =>
-            onSuccess(offlineSignedRoleStorage.saveTargetRole(namespace, signedPayload, repoId, checksum)) {
+            onSuccess {
+              // get the items before they get removed from the DB
+              targetItemRepo.findFor(repoId).flatMap { targetItems =>
+                offlineSignedRoleStorage.saveTargetRole(namespace, signedPayload, repoId, checksum)
+              }
+            } {
               case Valid((targetItems, newSignedRole)) =>
-                tufTargetsPublisher.newTargetsAdded(namespace, signedPayload.signed.targets, targetItems)
                 respondWithCheckSum(newSignedRole.checksum.hash) {
-                  complete(StatusCodes.NoContent)
+                  complete {
+                    for {
+                      _ <- deleteOutdated(targetItems, signedPayload.signed.targets.keys)
+                      _ <- tufTargetsPublisher.newTargetsAdded(namespace, signedPayload.signed.targets, targetItems)
+                    } yield StatusCodes.NoContent
+                  }
```

Comment:
I think this is getting quite complex.. Is there any way this can be a bit simplified? Probably extracting most of this code out of the routes?

I am actually not sure this would work, there are two targetItems in scope? or the first one is not in scope but is never actually used, so why get it?